src/videojs-http-streaming.js

```diff
@@ -789,7 +791,22 @@ class VhsHandler extends Component {
       },
       bandwidth: {
         get() {
-          return this.masterPlaylistController_.mainSegmentLoader_.bandwidth;
+          let bandwidthEst = this.masterPlaylistController_.mainSegmentLoader_.bandwidth;
+
+          const networkInformation = window.navigator.connection || window.navigator.mozConnection || window.navigator.webkitConnection;
+
+          if (this.options_.useNetworkInformationApi && networkInformation) {
+            // downlink returns Mbps
+            // https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/downlink
+            const effectiveBandwidthBitsPerSec = networkInformation.downlink * 1000 * 1000;
+
+            // downlink maxes out at 10 Mbps. In the event that the player estimates a bandwidth
+            // greater than 10 Mbps, use the larger value to ensure that high quality streams are not
+            // filtered out.
+            bandwidthEst = Math.max(bandwidthEst, effectiveBandwidthBitsPerSec);
```

Comment:
you mentioned that you noticed that `downlink` was also unreliable when throttling down? Not sure if it's worth mentioning that or not.

Reply:
The opposite:) I found that it’s more reliable than the player estimate even at low network bandwidths